src/averageAges.js

```diff
@@ -14,12 +14,28 @@
  *
  * @return {number}
  */
-function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+const MALE_SEX = 'm';
+const FEMALE_SEX = 'f';
+const ONE_CENTURY = 100;
+const calculateAgeSum = (data) => {
+  return data.reduce((averageAge, { born, died }) => {
+    const age = died - born;
+
+    return averageAge + age;
+  }, 0);
+};
+
+function calculateMenAverageAge(people, century = false) {
+  const men = people.filter(({ sex, died }) => {
+    const personCentury = Math.ceil(died / ONE_CENTURY);
+    const isSuitableCenruty = century ? century === personCentury : true;
+
+    return sex === MALE_SEX && isSuitableCenruty;
+  });
+
+  const menSumAge = calculateAgeSum(men);
+
+  return menSumAge / men.length;
```

Comment:
This logic - `/men.length` could also be moved to `calculateAgeSum` function(but renaming would be needed)